recipes/boost/all/conanfile.py

```diff
@@ -846,9 +846,22 @@ def _create_emscripten_libs(self):
                 self.output.info(cmd)
                 self.run(cmd)
 
+    @property
+    def _is_versioned_layout(self):
+        layout = self.options.get_safe("layout")
+        return layout == "versioned" or (layout == "b2-default" and os.name == 'nt')
+
     def package_info(self):
         gen_libs = [] if self.options.header_only else tools.collect_libs(self)
 
+        if self._is_versioned_layout:
+            version_tokens = str(self.version).split(".")
+            if len(version_tokens) >= 2:
+                major = version_tokens[0]
+                minor = version_tokens[1]
```

Comment:
All the versions we are generating with this recipe are `major.minor.patch`, we only need:

```
major, minor, _ = self.version.split(".")
```

Reply:
this may fail if there will be less tokens

Reply:
Sure, but all versions match that format.

https://github.com/conan-io/conan-center-index/blob/master/recipes/boost/config.yml

Reply:
I know, just wanted to be safe in case someone forks recipe and adds more versions in different format